File: gm-utils/src/main/java/src/commom/utils/comum/Box.java
package src.commom.utils.comum;

import src.commom.utils.object.Null;

public class Box<T> {
	
	private T value;
	
	public Box() {}
	
	public T get() {
		return this.value;
	}
	
	public void set(T valueP) {
		this.value = valueP;
	}
	
	public boolean isNotNull() {
		return !isNull();
	}
	
	public boolean isNull() {
		return Null.is(value);
	}
	
}

File: gm-utils/js/src/commom/utils/comum/Box.js
import Null from '../object/Null';

export default class Box {
	constructor(valueP) {
		this.set(valueP);
	}
	get() {
		return this.value;
	}
	set(valueP) {
		this.value = valueP;
	}
	isNotNull() {
		return !this.isNull();
	}
	isNull() {
		return Null.is(this.value);
	}
}
